### PR TITLE
Add manifests for deploying core etcd and etcd proxy

### PR DESCRIPTION
**Pull request by xmudrii**

### CODE REVIEW

File: artifacts/experimental/etcd.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Pod
+metadata:
+  name: etcd-pod
```
Comment (deads2k) [inline artifacts/experimental/etcd.yaml]: rename this.  It's the core-etcd, right?

File: artifacts/experimental/etcd-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  template:
+    metadata:
+      labels:
+        app: etcdproxy
+    spec:
+      containers:
+      - name: etcdproxy
+        image: quay.io/coreos/etcd:v3.2.18
+        command: ["/usr/local/bin/etcd", "grpc-proxy", "start"]
+        args: ["--endpoints=http://172.17.0.3:2379", "--namespace=/exp", "--listen-addr=0.0.0.0:23790"]
```
Comment (deads2k) [inline artifacts/experimental/etcd-proxy.yaml]: ip address is not known ahead of time, so this won't work generically.

File: artifacts/experimental/etcd.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    app: etcd
+spec:
+  containers:
+  - name: etcd
+    image: quay.io/coreos/etcd:v3.2.18
+    command: ["etcd"]
+    args: ["--listen-client-urls=http://0.0.0.0:2379", "--advertise-client-urls=http://0.0.0.0:2379"]
+    ports:
+     - name: etcd
+       containerPort: 2379
```
[the remark this answers is not in the document]
Comment (xmudrii) [reply artifacts/experimental/etcd.yaml]: The service is added in #2

File: artifacts/experimental/etcd-proxy.yaml (hunk 2 above)
Comment (deads2k) [reply artifacts/experimental/etcd-proxy.yaml]: reference the service, not the pod.

Comment (xmudrii) [reply artifacts/experimental/etcd-proxy.yaml]: Resorted to service in #2 https://github.com/xmudrii/etcdproxy-controller/pull/2/files#diff-2f3439ebbf3e4edae0f942275399111cR21

File: artifacts/experimental/etcd.yaml (hunk 1 above)
Comment (xmudrii) [reply artifacts/experimental/etcd.yaml]: Done

File: artifacts/experimental/etcd-proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  template:
+    metadata:
+      labels:
+        controller: epc
+    spec:
+      containers:
+      - name: etcdproxy
+        image: quay.io/coreos/etcd:v3.2.18
+        command: ["/usr/local/bin/etcd", "grpc-proxy", "start"]
+        args: ["--endpoints=http://etcd-svc-1.etcd.svc.cluster.local:2379", "--namespace=/exp", "--listen-addr=0.0.0.0:2379"]
```
Comment (deads2k) [inline artifacts/experimental/etcd-proxy.yaml]: nit: `.cluster.local` is configurable, so you probably want to end at `.svc`

File: artifacts/experimental/etcd.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  name: core-etcd
+  labels:
+    app: etcd
+spec:
+  containers:
+  - name: etcd
+    image: quay.io/coreos/etcd:v3.2.18
+    env:
+    - name: ETCD_CERT_FILE
+      value: "/etc/server-certs/server.pem"
```
Comment (deads2k) [inline artifacts/experimental/etcd.yaml]: this wasn't here before, right? Adding this means you have to add trust certs to have the proxy trust the core etcd, right?

**Comment by deads2k** [comment]
Looks good enough to me if you want to start here.  I'm not certain it runs though.

**Comment by xmudrii** [comment]
@deads2k Updated the PR to include client certs for etcd proxy. I have tested the manifests on my two clusters (one ran using `local-up-cluster.sh`, another running on DO), and the manifests work on both, so I guess they're okay.

**Pull request merged**